include/libtorrent/torrent_info.hpp

```diff
@@ -580,7 +573,7 @@ namespace libtorrent
 		// this is a copy of the info section from the torrent.
 		// it use maintained in this flat format in order to
 		// make it available through the metadata extension
-		boost::shared_array<char> m_info_section;
+		std::unique_ptr<char[]> m_info_section;
```

Comment:
I don't think the use of `span<>` in the interface is relevant, other than returning a span is problematic as it does not transfer ownership